app/components/provider_interface/user_list_card_component.rb

```diff
@@ -2,16 +2,18 @@ module ProviderInterface
   class UserListCardComponent < ViewComponent::Base
     include ViewHelper
 
-    attr_reader :provider_user, :full_name, :email_address
+    attr_reader :provider_user, :manageable_providers, :full_name, :email_address
 
-    def initialize(provider_user:)
+    def initialize(provider_user:, manageable_providers:)
       @provider_user = provider_user
+      @manageable_providers = manageable_providers
       @full_name = provider_user.full_name
       @email_address = provider_user.email_address
     end
 
     def providers_text
-      providers = provider_user.providers
+      providers = provider_user.providers.where(id: manageable_providers.pluck(:id)).order(:name)
```

Comment:
I think the `pluck` will run on each row render — could we push it up into the controller so it only runs once? Maybe the controller could generate structs or hashes like `{ user: user, providers: user.providers.where.etc }` and the component could be dumber

Reply:
I've done something like that here https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/2022/commits/d2e2a0ffaedccab44e33b08b12de803d3b8bd09f